### PR TITLE
add support for encrypting text files from web

### PR DESCRIPTION
**Pull request by Ninja3047**

### CODE REVIEW

File: web/src/render.tsx [hunk 1]
```diff
@@ -17,11 +17,56 @@
...
+    return (
+      <pre className='paste'>
+        <form class="hljs centered" onSubmit={this.handleSubmit}>
+          <textarea value={this.state.value} onChange={this.handleChange} />
+          <input type="submit" value="submit" />
+        </form>
+      </pre>
+    );
+  }
+}
```
Comment (edward-shen) [inline web/src/render.tsx]: I think this is the more modern way to write it?

```suggestion
class PasteForm extends React.Component {
  state = {
      value: "Sample text",
  };
  
  handleChange = () => (event) {
    this.setState({value: event.target.value});
  }

  handleSubmit = () => (event) {
    try {
      encrypt_string(this.state.value);
    } catch (e) {
      console.error(e);
    }

    event.preventDefault();
  }

  render = () => () {
    return (
      <pre className='paste'>
        <form class="hljs centered" onSubmit={this.handleSubmit}>
          <textarea value={this.state.value} onChange={this.handleChange} />
          <input type="submit" value="submit" />
        </form>
      </pre>
    );
  }
}
```

File: web/src/render.tsx [hunk 2]
```diff
@@ -17,11 +17,56 @@
...
+
+  handleChange(event) {
+    this.setState({value: event.target.value});
+  }
+
+  handleSubmit(event) {
+    try {
+      encrypt_string(this.state.value);
+    } catch (e) {
+      console.error(e);
```
Comment (edward-shen) [inline web/src/render.tsx]: Can we give context to the error message pls?

File: web/src/render.tsx [hunk 3]
```diff
@@ -17,11 +17,56 @@
...
+  }
+
+  render() {
+    return (
+      <pre className='paste'>
+        <form class="hljs centered" onSubmit={this.handleSubmit}>
+          <textarea value={this.state.value} onChange={this.handleChange} />
+          <input type="submit" value="submit" />
+        </form>
+      </pre>
```
Comment (edward-shen) [inline web/src/render.tsx]: Wait, huh? Do you need to hold state for this? I _swear_ you can access the textarea value from the event provided to handleSubmit

Also use the placeholder attribute for "Sample text" pls

**Review by edward-shen — CHANGES_REQUESTED** [review]

File: web/src/render.tsx (hunk 3 above)
Comment (Ninja3047) [reply web/src/render.tsx]: i think only onChange can get the textarea value
onSubmit is tied to the form so i think state is required

File: web/src/render.tsx (hunk 2 above)
Comment (Ninja3047) [reply web/src/render.tsx]: just removed the try/catch, error printing has been punted to rust

**Pull request merged**